src/Particle/CMakeLists.txt

```diff
@@ -53,6 +51,34 @@ target_include_directories(qmcparticle PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}")
 target_link_libraries(qmcparticle PRIVATE platform_cpu_LA)
 target_link_libraries(qmcparticle PUBLIC qmcnumerics qmcutil platform_runtime)
 
+# Workaround for broken OMP Target debug in clang as of 5d, main is more broken so
+# perhaps this problem is fixed.
+#
+# In the meantime we need to make a new target for just the omp target tainted files
+# so we don't have broken debug for all of ParticleSet.
+
+# this can be put back into qmcparticle when omp target isn't broken.
+set(PARTICLEOMPTARGET
+    createDistanceTableAAOMPTarget.cpp
+    createDistanceTableABOMPTarget.cpp)
+
+if(USE_OBJECT_TARGET)
+  add_library(qmcparticle_omptarget OBJECT ${PARTICLEOMPTARGET})
+else(USE_OBJECT_TARGET)
+  add_library(qmcparticle_omptarget ${PARTICLEOMPTARGET})
+endif(USE_OBJECT_TARGET)
+
+target_include_directories(qmcparticle_omptarget PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}")
+target_link_libraries(qmcparticle_omptarget PUBLIC qmcutil containers)
+
+if(ENABLE_OFFLOAD AND ${COMPILER} MATCHES "Clang")
+    target_compile_options(qmcparticle_omptarget PRIVATE $<$<CONFIG:DEBUG>:-O3>)
```

Comment:
Separating offload source organization can be useful also for other purpose.

I'm actually wondering if you ever want to debug with -O3. -O3 change significantly numerics and often ends up changing reproducibility. I usually do `-DCMAKE_CXX_FLAGS_DEBUG=-O3`.

Reply:
If you have a bug that only appears when optimization is on and you want all the other flags that come with Debug vs. Release with debug info it could be useful.

Reply:
Does the way you did end up with "-O3 -O0 -g" in the compile line?

My feeling is getting code passing tests "-O0" can be helpful but largely irrelevant. What we need is -O3 flawless right?